hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -161,6 +174,14 @@ private int readOneBlock(final byte[] b, final int off, final int len) throws IO
     if (off < 0 || len < 0 || len > b.length - off) {
       throw new IndexOutOfBoundsException();
     }
+    return 1; // 1 indicate success
+  }
+
+  private int readOneBlock(final byte[] b, final int off, final int len) throws IOException {
+    int validation = validate(b, off, len);
+    if (validation < 1) {
+      return validation;
```

Comment:
That would alter the existing flow. In certain cases it has to return 0 and -1 in some other cases, exceptions in case validations failed for len, off values.

Reply:
see comment above

Reply:
Done